shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -195,7 +197,7 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
         log.trace("Encoding message to send over the wire {}", msg);
         if (msg instanceof Append) {
             Append append = (Append) msg;
-            Session session = setupSegments.get(new SimpleImmutableEntry<>(append.segment, append.getWriterId()));
+            Session session = setupSegments.get(append.getSegment());
```

Comment:
This is incorrect behaviour. Different writers which write on the same segment have different sessions and need  to setup appends individually. This may result in data corruption.

Reply:
Got it. Thanks!